addons/account/models/account_move_line.py

```diff
@@ -1353,7 +1353,16 @@ def _sanitize_vals(self, vals):
                 vals.pop('debit', None)
                 vals.pop('credit', None)
             else:
-                vals['balance'] = vals.pop('debit', 0) - vals.pop('credit', 0)
+                move = self.env['account.move']
+                if 'move_id' in vals:
+                    move = self.env['account.move'].browse(vals['move_id'])
+                if move and not move.is_invoice(include_receipts=True) and vals.get('debit') is False and vals.get('credit') is False:
+                    vals['amount_currency'] = vals.get('amount_currency', 0.0)
+                    vals['balance'] = move.currency_id._convert(vals['amount_currency'], move.company_id.currency_id, move.company_id, move.date)
+                    vals['debit'] = vals['balance'] if vals['balance'] > 0.0 else 0.0
+                    vals['credit'] = -vals['balance'] if vals['balance'] < 0.0 else 0.0
```

Comment:
If we set the balance, we don't need to set the debit/credit and thus we should pop them like below and above.